cli/src/forge.rs

```diff
@@ -93,32 +95,42 @@ fn remove(root: impl AsRef<std::path::Path>, dependencies: Vec<Dependency>) -> e
     dependencies.iter().try_for_each(|dep| -> eyre::Result<_> {
         let path = libs.join(&dep.name);
         let git_mod_path = git_mod_libs.join(&dep.name);
-        println!("Removing {} in {:?}, (url: {}, tag: {:?})", dep.name, path, dep.url, dep.tag);
-
-        // remove submodule entry from .git/config
-        Command::new("git")
-            .args(&["submodule", "deinit", "-f", &path.display().to_string()])
-            .current_dir(&root)
-            .spawn()?
-            .wait()?;
-
-        // remove the submodule repository from .git/modules directory
-        Command::new("rm")
-            .args(&["-rf", &git_mod_path.display().to_string()])
-            .current_dir(&root)
-            .spawn()?
-            .wait()?;
-
-        // remove the leftover submodule directory
-        Command::new("git")
-            .args(&["rm", "-f", &path.display().to_string()])
-            .current_dir(&root)
-            .spawn()?
-            .wait()?;
-
-        // tell git to discard the removal of the submodule
-        Command::new("git").args(&["checkout", "--", "."]).current_dir(&root).spawn()?.wait()?;
-
+        
+        let answer = Question::new("To use this command your working tree must be clean.  All uncommitted changes will be lost.  Continue (y/N)?")
+        .default(Answer::NO)
+        .show_defaults()
+        .confirm();
+        
+        if answer == Answer::YES {
+            
+            println!("Removing {} in {:?}, (url: {}, tag: {:?})", dep.name, path, dep.url, dep.tag);
+
+            // remove submodule entry from .git/config
+            Command::new("git")
+                .args(&["submodule", "deinit", "-f", &path.display().to_string()])
+                .current_dir(&root)
+                .spawn()?
+                .wait()?;
+
+            // remove the submodule repository from .git/modules directory
+            Command::new("rm")
+                .args(&["-rf", &git_mod_path.display().to_string()])
+                .current_dir(&root)
+                .spawn()?
+                .wait()?;
+
+            // remove the leftover submodule directory
+            Command::new("git")
+                .args(&["rm", "-f", &path.display().to_string()])
+                .current_dir(&root)
+                .spawn()?
+                .wait()?;
+
+            // tell git to discard the removal of the submodule
+            Command::new("git").args(&["checkout", "--", "."]).current_dir(&root).spawn()?.wait()?;
```

Comment:
Instead of adding a prompt, maybe this can be changed to discard only the submodule change?

Something like this should work IIRC: `git checkout -- path/to/file`

Reply:
yeah, this shouldn't prompt.
I'm actually not sure what this even does.
imo if we rm via git rm we should leave up to the user to follow up on those changes

Reply:
In that case I suggest we go for feature parity with `install` by auto-committing only the removal of dependencies, and providing the `--no-commit` option to leave the changes unstaged in case user wants to manually handle it.

Reply:
that makes sense to make it consistent across all commands.
personally, I'm not a big fan of auto commits, but I don't have strong feelings about this here, guess that's just something dapp does :shrug:

Reply:
I'd also prefer for us to diverge here and not auto-commit, supportive of only removing the dependencies

Reply:
Ok ill update this to simple checkout the paths we want to ignore specifically.

Reply:
After looking at the comments from the issue https://github.com/gakonst/foundry/issues/673, Ive updated the PR to remove the "git checkout" invocation.  That command simply reverts changes in the working tree which is not the intention of forge remove.  Including the git checkout has the effect of leaving the submodule in the lib/ directory as well as reverting the removal of the submodule data from the .gitmodules file.  As such I have simply removed the command.